src/systems/systems/src/pneumatic/mod.rs

```diff
@@ -780,6 +781,78 @@ impl From<f64> for EngineModeSelector {
     }
 }
 
+pub struct PressureTransducer {
+    pressure_output: Option<Pressure>,
+
+    powered_by: ElectricalBusType,
+    is_powered: bool,
+}
+impl PressureTransducer {
+    pub fn new(powered_by: ElectricalBusType) -> Self {
+        Self {
+            pressure_output: None,
+            powered_by,
+            is_powered: false,
+        }
+    }
+
+    pub fn update(&mut self, context: &UpdateContext, container: &impl PneumaticContainer) {
+        self.pressure_output = if self.is_powered {
+            Some(container.pressure() - context.ambient_pressure())
+        } else {
+            None
+        };
```

Comment:
Maybe trying something new like this?
```suggestion
        self.pressure_output = self.is_powered.then_some(container.pressure() - context.ambient_pressure());
```